lib/handler/configurator/proxy.c

```diff
@@ -142,6 +149,71 @@ static int on_config_ssl_cafile(h2o_configurator_command_t *cmd, h2o_configurato
     return ret;
 }
 
+static h2o_cache_t *create_ssl_session_cache(uint64_t capacity, unsigned lifetime)
+{
+    return h2o_cache_create(H2O_CACHE_FLAG_MULTITHREADED, capacity, lifetime * 1000, h2o_socket_ssl_destroy_session_cache_entry);
```

Comment:
`lifetime * 1000` should be calculated using `uint64_t` (which is the type accepted by `h2o_cache_create). Otherwise, you are risking a integer overflow (that leads to undefined behavior).